src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -1108,7 +1108,7 @@ private bool ShouldContinueDispose(string name, DateTime? wakeupUtc, out int due
                 return true;
 
             // if we have a small value or even a negative one, simply don't dispose the database.
-            dueTime = (int)(wakeupUtc - DateTime.UtcNow).Value.TotalMilliseconds;
+            dueTime = (int)Math.Min(int.MaxValue, (wakeupUtc - DateTime.UtcNow).Value.TotalMilliseconds);
```

Comment:
This is not relevant directly to the issue but if the value is greater than 2G casting it to int will end with a negative value.
We use this value later for the timer and the max of a `Timer` value is 2G.